### PR TITLE
OCP4: Add unit test documentation

### PR DESCRIPTION
**Pull request by JAORMX**
This has been missing.

Signed-off-by: Juan Antonio Osorio Robles <jaosorior@redhat.com>

### CODE REVIEW

**Comment by lmzuccarelli** [comment]
Overall this looks good. I will leave the final lgtm for the team

File: docs/manual/developer/06_contributing_with_content.md [hunk 1]
```diff
@@ -1197,7 +1197,142 @@ See some of the existing testing code in `cmake/SSGCommon.cmake`.
...
+
+The unit testing approach for content requires a container
+where the scanning will be executed. This container will
+be a small replication of the environment where the content
+is intended to run.
+
+To build the container, from the project's root
+directory, execute the following command:
+
+    podman build --build-arg CLIENT_PUBLIC_KEY=(cat ~/.ssh/id_rsa.pub) -t ssg_test_suite -f Dockerfiles/test_suite-fedora
```
Comment (jhrozek) [inline docs/manual/developer/06_contributing_with_content.md:1211]: Is this line correct? I think first you meant to write `$(cat ~/.ssh/id_rsa.pub)`. But either way, I'm getting:
```
Error: accepts at most 1 arg(s), received 2
```
btw what worked for me was to first set the contents of the pubkey into a variable and then run:
```
podman build -t ssg_test_suite  -f Dockerfiles/test_suite-fedora --build-arg CLIENT_PUBLIC_KEY="${CLIENT_PUBLIC_KEY}"
```

File: docs/manual/developer/06_contributing_with_content.md [hunk 2]
```diff
@@ -1197,7 +1197,142 @@ See some of the existing testing code in `cmake/SSGCommon.cmake`.
...
+Remember this needs to be done every time before running the unit tests.
+
+Now, with all of this set, we'll run the unit test!
+
+    tests/test_rule_in_container.sh \
+        --dontclean --logdir logs_bash \
+        --remediate-using bash \
+        --name ssg_test_suite \
+        --datastream build/ssg-ocp4-ds.xml \
+        <rule name>
```
Comment (jhrozek) [inline docs/manual/developer/06_contributing_with_content.md:1332]: Can you double check this line, too, or give a complete example? I tried:
```
tests/test_rule_in_container.sh \
        --dontclean --logdir logs_bash \
        --remediate-using bash \
        --name ssg_test_suite \
        --datastream build/ssg-ocp4-ds.xml \
        route_ip_whitelist
```
and I'm getting:
```
INFO - The base image option has been specified, choosing Podman-based test environment.
INFO - Logging into logs_bash-3/test_suite.log
Traceback (most recent call last):
  File "/home/jhrozek/devel/compliance-as-code-content/tests/ssg_test_suite/test_env.py", line 485, in get_ip_address
    podman_output = subprocess.check_output(podman_cmd, stderr=subprocess.STDOUT)
  File "/usr/lib64/python3.9/subprocess.py", line 424, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib64/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['podman', 'inspect', 'time="2021-12-07T14:14:18+01:00" level=warning msg="Failed to decode the keys [\\"storage.options.ostree_repo\\"] from \\"/home/jhrozek/.config/containers/storage.conf\\"."\nti
me="2021-12-07T14:14:18+01:00" level=warning msg="Failed to decode the keys [\\"volume_path\\" \\"image_default_transport\\" \\"runtime\\" \\"conmon_path\\" \\"conmon_env_vars\\" \\"cgroup_manager\\" \\"init_path\\" \\"static_dir\\" \\"tmp
_dir\\" \\"max_log_size\\" \\"no_pivot_root\\" \\"cni_config_dir\\" \\"cni_plugin_dir\\" \\"infra_image\\" \\"infra_command\\" \\"enable_port_reservation\\" \\"label\\" \\"network_cmd_path\\" \\"num_locks\\" \\"EventsLogFilePath\\" \\"runt
imes\\" \\"runtimes.runc\\"] from \\"/home/jhrozek/.config/containers/containers.conf\\"."\ntime="2021-12-07T14:14:18+01:00" level=warning msg="Failed to decode the keys [\\"volume_path\\" \\"image_default_transport\\" \\"runtime\\" \\"con
mon_path\\" \\"conmon_env_vars\\" \\"cgroup_manager\\" \\"init_path\\" \\"static_dir\\" \\"tmp_dir\\" \\"max_log_size\\" \\"no_pivot_root\\" \\"cni_config_dir\\" \\"cni_plugin_dir\\" \\"infra_image\\" \\"infra_command\\" \\"enable_port_res
ervation\\" \\"label\\" \\"network_cmd_path\\" \\"num_locks\\" \\"EventsLogFilePath\\" \\"runtimes\\" \\"runtimes.runc\\"] from \\"/home/jhrozek/.config/containers/containers.conf\\"."\ntime="2021-12-07T14:14:18+01:00" level=warning msg="F
ailed to decode the keys [\\"volume_path\\" \\"image_default_transport\\" \\"runtime\\" \\"conmon_path\\" \\"conmon_env_vars\\" \\"cgroup_manager\\" \\"init_path\\" \\"static_dir\\" \\"tmp_dir\\" \\"max_log_size\\" \\"no_pivot_root\\" \\"c
ni_config_dir\\" \\"cni_plugin_dir\\" \\"infra_image\\" \\"infra_command\\" \\"enable_port_reservation\\" \\"label\\" \\"network_cmd_path\\" \\"num_locks\\" \\"EventsLogFilePath\\" \\"runtimes\\" \\"runtimes.runc\\"] from \\"/home/jhrozek/
.config/containers/containers.conf\\"."\ntime="2021-12-07T14:14:18+01:00" level=warning msg="Failed to decode the keys [\\"volume_path\\" \\"image_default_transport\\" \\"runtime\\" \\"conmon_path\\" \\"conmon_env_vars\\" \\"cgroup_manager
\\" \\"init_path\\" \\"static_dir\\" \\"tmp_dir\\" \\"max_log_size\\" \\"no_pivot_root\\" \\"cni_config_dir\\" \\"cni_plugin_dir\\" \\"infra_image\\" \\"infra_command\\" \\"enable_port_reservation\\" \\"label\\" \\"network_cmd_path\\" \\"n
um_locks\\" \\"EventsLogFilePath\\" \\"runtimes\\" \\"runtimes.runc\\"] from \\"/home/jhrozek/.config/containers/containers.conf\\"."\n3041947431caa6c7e643ac0f2cb08aed59cb997e84f6aed3311f8a87dd22ca36', '--format', '{{.NetworkSettings.IPAdd
ress}}']' returned non-zero exit status 125.
```

**Review by jhrozek — COMMENTED** [review]
The docs are good, but we should double check the examples

File: docs/manual/developer/06_contributing_with_content.md (hunk 1 above)
Comment (JAORMX) [reply docs/manual/developer/06_contributing_with_content.md]: oops! Sorry, this was `fish` format, not bash

File: docs/manual/developer/06_contributing_with_content.md (hunk 2 above)
Comment (JAORMX) [reply docs/manual/developer/06_contributing_with_content.md:1332]: Just tried it:
```
tests/test_rule_in_container.sh --dontclean --logdir logs_bash --remediate-using bash --name ssg_test_suite --datastream build/ssg-ocp4-ds.xml route_ip_whitelist
Setting console output to log level INFO
INFO - The base image option has been specified, choosing Podman-based test environment.
INFO - Logging into logs_bash/test_suite.log
ERROR - No tests found matching the rule ID(s) 'route_ip_whitelist'
WARNING - Nothing has been tested!
```

**Review by Vincent056 — COMMENTED** [review]
This looks good, wondering if we want to add some information for tests of rules that uses jq filter

**Comment by JAORMX** [comment]
@Vincent056 let's add that in a subsequent PR. Now that you have knowledge about that, could you do it?

**Comment by Vincent056** [comment]
> @Vincent056 let's add that in a subsequent PR. Now that you have knowledge about that, could you do it?

Sure thing!

**Review by Vincent056 — APPROVED** [review]
/lgtm

**Pull request merged**